projects/plugins/boost/app/features/optimizations/render-blocking-js/class-render-blocking-js.php

```diff
@@ -198,17 +198,25 @@ protected function get_script_tags( $buffer ) {
 	}
 
 	/**
-	 * Adds the ignore attribute to scripts inside HTML comments.
+	 * Adds the ignore attribute to scripts in the exclusion list.
 	 *
 	 * @param string $buffer Captured piece of output buffer.
 	 *
 	 * @return string
 	 */
-	protected function ignore_scripts_in_comments( $buffer ) {
-		return preg_replace_callback(
+	protected function ignore_exclusion_scripts( $buffer ) {
+		$exclusions = array(
+			// Scripts inside HTML comments.
 			'~<!--.*?-->~si',
-			function ( $comment_match ) {
-				return str_replace( '<script', sprintf( '<script %s="%s"', esc_html( $this->ignore_attribute ), esc_attr( $this->ignore_value ) ), $comment_match[0] );
+
+			// Scripts with application/json type
+			'~<script.*type="application/json".*>.*</script>~si',
```

Comment:
The HTML spec allows single-quotes or no quotes here.

You can detect these cases using a regexp like this:
```suggestion
			'~<script.*type=(?<q>["\']*)application/json\k<q>*>.*</script>~si',
```

Reply:
Thanks for the suggestion. Updated this now.